src/core/api.py

```diff
@@ -12,30 +12,52 @@
 
 
 def handle_graphql_response(func):
+    """Decorator that handles graphql responses and error handling"""
+
     def parse_data_and_errors_from_gql_response(*args, **kwargs):
         result = func(*args, **kwargs)
 
         if result.get("errors"):
             errors = result.get("errors")
             raise Exception(errors)
 
+        data = None
         if result.get("data"):
             # Data is not null
             data = result.get("data")
 
             # If there was only one resolver requested, shortcut and give to the
             # user
-            while len(data) == 1:
-                try:
-                    data = next(iter(data.values()))
-                except Exception:
-                    break
+            data = access_next_data_key(data)
 
         return data
 
     return parse_data_and_errors_from_gql_response
 
 
+def access_next_data_key(data: dict):
+    """Helper function to access the first key of a dict if it's the only thing
+    in there. Useful for graphql resolver returns."""
+
+    if len(data) == 1 and type(data) == dict:
+        nextkey = next(iter(data.keys()))
+        data = next(iter(data.values()))
+        log.debug(f"diving into {nextkey} object on data object")
```

Comment:
Can't this be written like this:

```
if len(data) == 1:
        nextkey, data = next(iter(data.items()))
        log.debug(f"diving into {nextkey} object on data object")
```
        
Since `data` is already specified as a dict, wouldn't it throw an `TypeError`?
And also `data.items()` can be used to make it a one liner?

Reply:
Yeah you're right, the code evolved organically at two separate times so as of writing it made sense, but from a clean perspective this is silly. Will fix.